vendor/scripts/spade.js

```diff
@@ -0,0 +1,230 @@
+var Spade = function Spade() {
+	this.stack = [];
+}
+Spade.prototype = {
+	track: function(_elem) {
+		this.target = _elem;
+
+		var spade = this;
+
+		var el = document.createElement("div");
+
+		keyHook = null;
+		if(_elem.textInput && _elem.textInput.getElement) {
+			keyHook = _elem.textInput.getElement();
+		} else {
+			keyHook = _elem;
+		}
+		keyHook.addEventListener("keydown", function(_event) {spade.createEvent(spade.target)});
+		//Maybe this is needed depending on Firefox/other browsers? Duplicate non-diff events get compiled down.
+		keyHook.addEventListener("keyup", function(_event) {spade.createEvent(spade.target)});
+
+		_elem.addEventListener("mouseup", function(_event) {spade.createEvent(spade.target)});
+		
+	},
+	createEvent: function(_target) {
+		if(_target.getValue) {
+			this.stack.push({
+				"startPos":_target.selection.getCursor(),
+				"endPos":_target.selection.getSelectionAnchor(),
+				"content":_target.getValue(),
+				"timestamp":(new Date()).getTime()
+			});
+		} else {
+			this.stack.push({
+				"startPos":_target.selectionStart,
+				"endPos":_target.selectionEnd,
+				"content":_target.value,
+				"timestamp":(new Date()).getTime()
+			});
+		}
+	},
+	clear: function() {

```

Comment:
This is never called? Not sure why it is needed.

Also, we need a `destroy()` or `untrack()` method that we can call to remove our key and mouse listeners and possibly the playback timeout if we are playing, and then to call this when we stop tracking in SpellView or when SpellView is destroyed.